session-manager/src/session.rs

```diff
@@ -12,63 +12,130 @@
 //! information on licensing and copyright.
 
 use crate::error::SessionManagerError;
+use mbedtls::alloc::List;
 use policy_utils::principal::Identity;
 
 use std::{
     io::{Read, Write},
+    sync::{Arc, Mutex},
     vec::Vec,
 };
 
-use rustls::{Certificate, ServerConnection};
-
 ////////////////////////////////////////////////////////////////////////////////
 // Sessions.
 ////////////////////////////////////////////////////////////////////////////////
 
 /// A principal is an individual identified with a cryptographic certificate,
 /// and assigned a set of roles that dictate what that principal can and cannot
 /// do in a Veracruz computation.
-pub type Principal = Identity<Certificate>;
+pub type Principal = Identity<List<mbedtls::x509::Certificate>>;
```

Comment:
Do we want to `use mbedtls::x509::Certificate` at the top of the file to make this simpler?

For me, it can go either way. However, there has been the trend within the project (and I think it's a preference within the rust community) to use a `use` statement instead of having long lists of modules, submodules, etc. within the code.

Again, this is a question, not a demand, because for me, it can go either way

Reply:
I think that's worth changing. I'll do that.